### ISSUE TITLE
SP Topology is weird when input width and height are different

### ISSUE THREAD

**Issue by vladim0105** [htm-community/htm.core#961]
I’ve been experimenting with topology and the spatial pooler, but I am getting strange results. My input consists of an SDR which has 3 filled circles of active bits. I expect the spatial pooler to produce output which has a similar topology, however what I am getting instead is this:
![image](https://user-images.githubusercontent.com/13138060/142458784-8d128007-aa4d-4bfb-86d8-e37e9f6f15c8.png)
The output matches the y-location of the input, but for some reason it has activations along the entire width of the column dimensions. It seems to be wrapping around the width. This happens even when the spatial pooler column dimensions match the input space dimensions. Is this intended behavior?

**Comment by ctrl-z-9000-times** [comment]
Ok, I think I reproduced the issue here.

Setting the spatial pooler parameter: `wrapAround=False` appears to fix it.

**Comment by vladim0105** [comment]
The issue persisted for me even when wrapAround was set to false, forgot to include that in the issue :/

**Comment by ctrl-z-9000-times** [comment]
Could you post your spatial pooler's parameters?
It would help in reproducing your issue.

**Comment by vladim0105** [comment]
Yes of course, I've changed them since I moved on to 1:1 aspect ratio input, but they were something like this:
```
    sp_args.inputDimensions = (512, 288)
    sp_args.columnDimensions = (128, 64)
    sp_args.potentialPct = 0.25
    sp_args.potentialRadius = 10
    sp_args.localAreaDensity = 0.1
    sp_args.globalInhibition = False
    sp_args.wrapAround = False
    sp_args.synPermActiveInc = 0.001
    sp_args.synPermInactiveDec = 0.00001
    sp_args.stimulusThreshold = 1
    sp_args.boostStrength = 0
    sp_args.dutyCyclePeriod = 10000000
```
Don't hesitate to ask if you need more info! :)

**Comment by ctrl-z-9000-times** [comment]
Sorry, I don't know when I will have the time to work on this issue.
